### PR TITLE
Fix previous source map not being used correctly.

### PR DESCRIPTION
**Pull request by izaakschroeder**
The previous source map should be applied in `process`, not in `toResult` even though the documentation says `toResult` supports these properties. See: https://github.com/postcss/postcss/blob/master/docs/api.md#roottoresultopts for information on this API. Additional tests have been added to verify this behavior, including the addition of a more complex example using the `less` CSS preprocessor.

Fixes https://github.com/metalabdesign/css-split-webpack-plugin/issues/4.

There is still an issues involving the source filenames within the source map but that issue does not appear to be related. Namely that https://github.com/webpack/css-loader/blob/7d2abbaa82a093fa9b71223e1b5ab485cf7e9623/lib/loader.js#L106 injects `webpack://` into the sourcemap source list _always_, and then on the second pass when everything gets emitted `devtoolModuleFilenameTemplate` gets called, producing things like: `webpack://webpack://styles.less`. This issue occurs irrespective of whether `CSSSplitWebpackPlugin` is enabled or not. See https://github.com/webpack/css-loader/issues/280 for additional information.

### CODE REVIEW

**Comment by nealgranger** [comment]
This will go.

File: test/spec/index.spec.js [hunk 1]
```diff
@@ -17,75 +20,114 @@ const config = (options) => {
...
       expect(stats).to.not.be.null;
       expect(stats.assetsByChunkName).to.have.property('main')
         .to.contain('styles.css');
-    });
-  });
+    })
+  );
   it('should give sensible names by default', () => {
-    return webpack({size: 3, imports: true, preserve: true}).then((stats) => {
+    return webpack({size: 3, imports: true, preserve: true}).then(({stats}) => {
```
Comment (jamesdphillips) [inline test/spec/index.spec.js]: Sort of curious why this test didn't fail before

**Comment by jamesdphillips** [comment]
itsagooda

**Pull request merged**